applications/compareSummary.cpp

```diff
@@ -0,0 +1,365 @@
+/*
+  Copyright 2016 Statoil ASA.
+  This file is part of the Open Porous Media project (OPM).
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "compareSummary.hpp" 
+
+int main (int argc, char ** argv){
+  if(argc != 5){
+    printHelp();
+    return 0;
+  }
+ 
+  const char * smspecFile1 = argv[1];
+  const char * unsmryFile1 = argv[2];
+  const char * smspecFile2 = argv[3];
+  const char * unsmryFile2 = argv[4];
+  double relative_tolerance_max = 1 ;
+  double relative_tolerance_median_max = 0.1;
+  stringlist_type * file1 = stringlist_alloc_new();
+  stringlist_type * file2 = stringlist_alloc_new();
+  stringlist_append_copy(file1, unsmryFile1);
+  stringlist_append_copy(file2, unsmryFile2);
+ 
+
+  SummaryReader read(relative_tolerance_max,relative_tolerance_median_max); //may also use sat function for tolerance limits
```

Comment:
this needs to be in a try-catch block. you are throwing in the reader, you need to catch those exceptions (and report / graciously handle the failure).